utils/src/main/java/org/apache/cloudstack/utils/security/DigestHelper.java

```diff
@@ -93,4 +95,33 @@ public static boolean isAlgorithmSupported(String checksum) {
         }
         return true;
     }
+
+    /**
+     * Checks if the algorithm prefix is provided on the checksum
+     */
+    public static void checksumSanity(String checksum) {
+        if(StringUtils.isNotEmpty(checksum)) {
+            if (checksum.contains("{") && checksum.contains("}")) {
+                int s = checksum.indexOf('{');
+                int e = checksum.indexOf('}');
+                if (s == 0 && e > s+1) { // we have an algorithm name of at least 1 char
+                    String algorithm = checksum.substring(s + 1, e);
+                    Map<String, Integer> map = creatPaddingLengths();
```

Comment:
Can you please rename the method `creatPaddingLengths`? It is misleading (at least to me).
This method actually returns a map between hashing methods and their respective hash size. What about ``getMapHashTypeBySize` or `getMapHashByDigestSize`, or something that can express more clearly what the method is doing.